Context.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+const EventEmitter = require('events');
+const IncomingWebSocket = require('./IncomingWebSocket');
+const OutgoingWebSocket = require('./OutgoingWebSocket');
+const logger = require('./loggerFactory');
+const { createTarget } = require('./util');
+const {
+  kSender,
+  kUpstreamClosed,
+  kStartConnection,
+  kMessageReceived,
+  kError,
+  kSendMessage,
+  kQueueMessage,
+  kDrainMessage,
+  kConnectionOpened,
+  kDequeueMessage,
+  kClientClosed,
+  kCleanup,
+  kDrainCompleted,
+  kReleaseTap,
+  kAddNewContext,
+  kUpstreamRestart,
+  config,
+  INCOMING,
+  OUTGOING
+} = require('./constants');
+
+/**
+ * Context holds the information for the incoming and outgoing sockets.
+ *
+ * Instantiates the incoming and outgoing sockets and registers the listeners.
+ * Acts as a bridge between incoming and outgoing sockets.
+ */
+class Context extends EventEmitter {
+
+  /**
+   * Creates the context with connection identifier.
+   *
+   * @param {string} connectionId
+   */
+  constructor(connectionId) {
```

Comment:
Not necessarily since in the starting, we do not have the connectionId with us and latter assigned from the backend system.

Reply:
should this be removed? 
& the assign `this.connectionId = null`?